datalad/distribution/subdatasets.py

```diff
@@ -112,20 +102,16 @@ def _parse_git_submodules(dspath, recursive):
     except CommandError as e:
         raise InvalidGitRepositoryError(exc_str(e))
 
-    for line in stdout.split('\n'):
-        if not line:
+    for line in stdout.split('\0'):
+        if not line or not line.startswith('160000'):
             continue
         sm = {}
-        sm['state'] = status_map[line[0]]
-        props = submodule_full_props.match(line[1:])
-        if props:
-            sm['revision'] = props.group(1)
-            sm['path'] = opj(dspath, props.group(2))
-            sm['revision_descr'] = props.group(3)
-        else:
-            props = submodule_nodescribe_props.match(line[1:])
-            sm['revision'] = props.group(1)
-            sm['path'] = opj(dspath, props.group(2))
+        props = submodule_full_props.match(line)
+        sm['revision'] = props.group(2)
+        subpath = opj(dspath, props.group(4))
+        sm['path'] = subpath
+        if not exists(subpath) or not GitRepo.is_valid_repo(subpath):
```

Comment:
No, not in this context, as this is no type reporting. The hit is listed as a submodule, but there is not repo. Hence the submodule is absent. Likely, it is an empty submodule "mountpoint" directory. but it could be a file (typechange). Regardless, the submodule is not present.